jetson/nav/searchPointGeneration.py

```diff
@@ -0,0 +1,78 @@
+import math
+import numpy as np
+import matplotlib.pyplot as plt
+
+def generateSpiralSearchPoints(radius, sides, coils):
+    awayStep = radius/sides
+    aroundStep = coils/sides
+    aroundRadians = aroundStep * 2 * math.pi
+    coordinates = [(0,0)]
+    for i in range(1, sides+1):
+        away = i * awayStep
+        around = i * aroundRadians
+        x =  math.cos(around) * away
+        y = math.sin(around) * away
+        coordinates.append((x,y))
+    return coordinates
+    
+def generateEquidistantSpiralSearchPoints(radius, distance, coils, rotation):
+    thetaMax = coils * 2 * math.pi
+    awayStep = radius / thetaMax
+    theta = distance / awayStep
+    coordinates = [(0,0)]
+    while theta <= thetaMax:
+        away = awayStep * theta
+        around = theta + rotation
+        x = math.cos(around) * away
+        y = math.sin(around) * away
+        coordinates.append((x,y))
+        theta += distance / away
+    return coordinates
+
+def generateSquareSpiral (points, distance):
+    directions = [(0,1), (1,0), (0,-1), (-1,0)]
+    coordinates = [(0,0)]
+    new_distance = distance
+    for i in range(0,points):
+        coordinates.append( (coordinates[-1][0]+new_distance*directions[i%4][0], coordinates[-1][1]+new_distance*directions[i%4][1]) )
+        new_distance += (i%2)*distance
+    return coordinates
+
+def showCoords(coordinates):
+    x_coords = [i[0] for i in coordinates]
+    y_coords = [i[1] for i in coordinates]
+    plt.scatter(x_coords, y_coords)
+    for i in range(1, len(x_coords)):
+        plt.plot(x_coords[i-1:i+1], y_coords[i-1:i+1], 'ro-')
+    plt.show()
+
+def cart2pol(x, y):
+    rho = np.sqrt(x**2 + y**2)
+    phi = -1 * (np.arctan2(y, x) * 180 / math.pi - 90)
+    return(rho, phi)
+
+print("\n-----Search types-----\nRadially Equidistant Spiral: 0\nPoint Equidistant Spiral: 1\nSquare Spiral: 2\n")
+search_type = input("Select a search type: ")
+if search_type == '0': # Point Equidistant Spiral
+    # generateSpiralSearchPoints 
+    #           (Radius of spiral, Number of Points, Number of coils)
+    coords = generateSpiralSearchPoints(20, 200, 10)
```

Comment:
Yes, but it would mean that we would need to standardize the parameters to the different searches. Otherwise, we can also just ask for input depending on the search type that is chosen.

Reply:
@cgiger00 Since we have multiple search types, there are multiple types of search paths that can be generated. Each of these has different parameters so it is a little hard to take the parameters as inputs to the program. One possible solution would be to prompt the user to specify these parameters depending on the search type that they choose. Otherwise we could specify locations to change these parameters within the file itself. Do you have any way which you would prefer?

Reply:
yes i think prompting the user will be the cleanest way to go about this